src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs

```diff
@@ -234,7 +236,7 @@ public RestoreCommand(FrameworkName fallbackFramework)
                 Reports.Information.WriteLine("Updating the invalid lock file with {0}",
                     "dnu restore --lock".Yellow().Bold());
                 useLockFile = false;
-                Lock = true;
+                relock = true;
```

Comment:
I didn't look deeply but I suspect this is the key line that fixes the problem? It is expected that this Lock property should only ever be set by the command line and not by other invalid lock files.